transport/tchannel/header.go

```diff
@@ -36,14 +36,14 @@ import (
 
 // pullBaggage pulls the context headers from the given transport.Headers,
 // deleting them from the original headers map.
-func pullBaggage(headers transport.Headers) transport.Headers {
-	ctxHeaders := make(transport.Headers)
+func pullBaggage(headers transport.Headers) map[string]string {
```

Comment:
Can you explain why you're using `map[string]string` here (and in several other places) instead of a defined type?